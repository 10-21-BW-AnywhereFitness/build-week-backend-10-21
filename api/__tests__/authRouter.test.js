const request = require('supertest')
const server = require('../server')
const db = require('../data/db-config')

beforeAll(async () => {
  await db.migrate.rollback()
  await db.migrate.latest()
})
beforeEach(async () => {
  await db.seed.run()
})
afterAll(async () => {
  await db.destroy()
})

it('sanity check', () => {
  expect(true).not.toBe(false)
})

describe('server.js', () => {
  it('is the correct testing environment', async () => {
    expect(process.env.NODE_ENV).toBe('testing')
  })
})

describe('[POST]  /api/auth/register', () => {
  let res
  beforeEach(async () => {
    res = await request(server).post('/api/auth/register').send({ username: "taco", password: "12345" })
  })

  it('responds with 201 created', () => {
    expect(res.status).toBe(201)
  })
  it('causes a user to be added to the db', async () => {
    const users = await db('users')
    expect(users.length).toBe(5)
  })
  it('responds with newly created user', () => {
    expect(res.body).toMatchSnapshot()
  })
  it('new user has user_id, username, and role_id', () => {
    expect(res.body).toMatchObject({
      user_id: 5,
      username: "taco",
      role_id: 2
    })
  })
})

describe('[POST]  /api/auth/login', () => {
  let res 
  beforeEach(async () => {
    res = await request(server).post('/api/auth/login').send({ username: "selene", password: "12345" })
  })
  it('responds with a status 200', () => {
    expect(res.status).toBe(200);
  })
  it('sends back user_id, welcome message, and token', () => {
    expect(res.body).toMatchObject({
      "user_id": 4,
      "message": "Welcome back selene"
    })
    expect(res.body.token).toBeDefined()
  })
})

