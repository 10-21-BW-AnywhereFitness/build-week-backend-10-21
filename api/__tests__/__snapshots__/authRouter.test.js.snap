// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[POST]  /api/auth/register responds with newly created user 1`] = `
Object {
  "role_id": 2,
  "user_id": 5,
  "username": "taco",
}
`;
